test/core/crawlers/cheerio_crawler.test.ts

```diff
@@ -252,16 +252,18 @@ describe('CheerioCrawler', () => {
     });
 
     test('should serialize body and html', async () => {
-        expect.assertions(2);
+        expect.assertions(3);
         const sources = [`${serverAddress}/special/html-type`];
         const requestList = await RequestList.open(null, sources);
 
         const cheerioCrawler = new CheerioCrawler({
             requestList,
             maxRequestRetries: 0,
             maxConcurrency: 1,
-            requestHandler: ({ $, body }) => {
+            requestHandler: ({ $, body, request }) => {
                 expect(body).toBe(responseSamples.html);
+                // test that `request.loadedUrl` is no longer optional
+                expect(request.loadedUrl.length).toBe(sources[0].length);
```

Comment:
1. does the comment have to be there? it makes sense in the context of this PR, but later on, I think it won't be necessary
2. why do you need to check the length instead of just comparing the string values?

Reply:
the test checks if `loadedUrl` is typed to `string` and not `string | undefined` which the latter would fail to compile here. you wont verify that if you compare the value, you need to call something on the value. the comment is there to explain this.

Reply:
Well, then the comment failed :smile: Why not check `typeof` then?

Reply:
That's the very same story, the test is about type level, not runtime values, those were correct already.

Reply:
Ah, I get it now. I guess there's no way to do `expect`, but on the type level. Could we reword the comment to something like "Check that `request.loadedUrl` cannot not undefined. If it could, accessing the `length` property would cause a Typescript error."?

Reply:
there are ways to test on type level too, often its about making assignments to explicitly typed variables (combined with using `@ts-expect-error` for cases that should fail). we could do the same here too. or we could use something like https://github.com/dsherret/conditional-type-checks

few examples of such tests here https://github.com/mikro-orm/mikro-orm/blob/master/tests/types.test.ts

Reply:
`conditional-type-checks` is pretty cool, if you think it's worth adopting it for cases like this, I'm in favor